pczt/src/common.rs

```diff
@@ -0,0 +1,51 @@
+use std::collections::BTreeMap;
+
+/// Global fields that are relevant to the transaction as a whole.
+#[derive(Clone)]
+pub(crate) struct Global {
```

Comment:
Keystone's proposal included a `network` field (indicating mainnet vs testnet vs ?) at the global level, possibly because in my 2019 PCZT draft I suggested it. However, I note that PSBTs (BIP 174) do not include such a field (AFAICT). Is such a field necessary, and/or useful?

Reply:
I don't see why it would be necessary or useful here, but it looks like BIP 174 does include an implicit indication of the network in the serialized extended public keys.

Reply:
We meet a similar problem on Unisat's Fractal Bitcoin recently. PSBT doesn't have a network field, so we have to set it on the global map field which means that becomes a custom protocol between Unisat and us. 
This can be useful if the chain have multiple sub/child chains like Bitcoin vs Litecoin or Dogecoin. In theory Litecoin can also use PSBT,  but without network field, we cannot know some accurate info like coin precision.
Anyway, that's depends on Zcash's needs.

Reply:
If a Derivation Path is provided wouldn't that account for having `network` field?

Reply:
Hmm, this is a good point. And as @arya2 notes above and @pacu notes below, we will already be semi-explicitly encoding the network into the PCZT at some later stage. So I think it makes sense for the Creator to encode the network up-front, and then every later instance of data that includes network information can be checked against it for correctness.

Next question then is what format this network field should have. A SLIP 44 `coin_type` (as used in BIP 32 and ZIP 32 derivation paths) would make sense for mainnet applications, but SLIP 44 only defines a single testnet code for all coins. Would that be sufficient for Zcash (and potentially other) testnet uses? (I think so, because if you're running against testnet then you already likely know what network you're dealing with, and the important part is that we have domain separation between mainnet and non-mainnet, as well as between Zcash and non-Zcash mainnets.)

Reply:
I decided to add a SLIP 44 `coin_type` field.